File: Ahorn/entities/power_refill.jl
module BatteriesPowerRefill

using ..Ahorn, Maple

@mapdef Entity "batteries/power_refill" PowerRefill(x::Integer, y::Integer, oneUse::Bool=false)

const placements = Ahorn.PlacementDict(
    "Battery Power Refill" => Ahorn.EntityPlacement(
        PowerRefill
    ),
)

spriteOneDash = "batteries/power_refill/idle00"

function getSprite(entity::PowerRefill)
    return spriteOneDash
end

function Ahorn.selection(entity::PowerRefill)
    x, y = Ahorn.position(entity)
    sprite = getSprite(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::PowerRefill, room::Maple.Room)
    sprite = getSprite(entity)
    Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/power_refill.lua
local utils = require("utils")

local power_refill = {}

power_refill.name = "batteries/power_refill"
power_refill.depth = -100

power_refill.placements = {
    name = "default",
    data = {
        oneUse = false
    }
}

power_refill.texture = "batteries/power_refill/idle00"

function power_refill.selection(room, entity)
    return utils.rectangle(entity.x - 6, entity.y - 5, 11, 11)
end

return power_refill
